### PR TITLE
🧞 Generic sign modal

### PR DESCRIPTION
**Pull request by WRadoslaw**
- #2204

### CODE REVIEW

File: packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx [hunk 1]
```diff
@@ -263,25 +263,36 @@ export const AnnounceCandidacyModal = () => {
...
+            <TextMedium>
+              Also you intend to stake <TokenValue value={form.watch('staking.amount') ?? BN_ZERO} />.
+            </TextMedium>
+          </>
+        }
         transaction={announceCandidacyTransaction}
         signer={activeMember.controllerAccount}
-        stake={form.watch('staking.amount') ?? BN_ZERO}
+        onClose={hideModal}
         service={state.children.announceCandidacyTransaction}
```
Comment (thesan) [inline packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx:279]:
```suggestion
```
Can't these 2 always be accessed from inside `SignTransactionModal` with:
```ts
  const { modal, hideModal, currentModalMachine, showModal, modalData } = useModal()
  const [service] = useMachine(currentModalMachine)
```
?

File: packages/ui/src/memberships/modals/TransferInviteModal/TransferInviteModal.tsx [hunk 2]
```diff
@@ -62,13 +63,21 @@ export function TransferInviteModal() {
 
   if (state.matches('transaction')) {
     const { targetMember, numberOfInvites } = state.context
+    const transaction = api?.tx?.members?.transferInvites(member.id, targetMember.id, numberOfInvites)
+    const plural = numberOfInvites.gt(new BN(1))
+    const handle = targetMember.handle
     return (
-      <TransferInviteSignModal
+      <SignTransactionModal
+        buttonText="Sign and Send"
```
Comment (thesan) [inline packages/ui/src/memberships/modals/TransferInviteModal/TransferInviteModal.tsx:71]:
```suggestion
```
"Sign and Send" could be the default

File: packages/ui/src/common/modals/SignTransactionModal/SignTransactionModal.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import { TextMedium, TokenValue } from '@/common/components/typography'
+import { BorderRad, Colors, Sizes } from '@/common/constants'
+import { useSignAndSendTransaction } from '@/common/hooks/useSignAndSendTransaction'
+import { TransactionModal, TransactionModalProps } from '@/common/modals/TransactionModal'
+import { Address } from '@/common/types'
+
+export interface SignTransactionModalProps extends Omit<TransactionModalProps, 'children'> {
+  additionalTransactionInfo?: TransactionInfoProps[]
+  buttonText: string
+  textContent: React.ReactElement
```
Comment (thesan) [inline packages/ui/src/common/modals/SignTransactionModal/SignTransactionModal.tsx:25]: Please use the `children` prop for this

**Review by thesan — CHANGES_REQUESTED** [review]
Great refactor :tophat: !

I have just a few suggestion and questions:

File: packages/ui/src/bounty/modals/AuthorizeTransactionModal/AuthorizeTransactionModal.tsx [hunk 4]
```diff
@@ -27,7 +27,7 @@ export interface Props {
   useMultiTransaction?: MultiTransactionConfig
   skipQueryNodeCheck?: boolean
 }
-
+// todo should be removed
```
[the remark this answers is not in the document]
Comment (WRadoslaw) [reply packages/ui/src/bounty/modals/AuthorizeTransactionModal/AuthorizeTransactionModal.tsx:30]: Future PR, it was decided to omit Bounty module for now.

File: packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx (hunk 1 above)
Comment (WRadoslaw) [reply packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx:279]: I guess `onClose` can be removed, but service names differ from modal to modal.

**Review by thesan — APPROVED** [review]
image :heart_eyes:

**Pull request merged**